File: Sample Apps /SampleAppWakeUp/app/src/main/java/com/mist/sample/samplewakeup/BootReceiver.java
package com.mist.sample.samplewakeup;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;
import com.mist.sample.samplewakeup.application.MainApplication;
import com.mist.sample.samplewakeup.util.AltBeaconUtil;

public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = "SampleWakeUpBroadcastReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        switch (intent.getAction()) {
            case Intent.ACTION_BOOT_COMPLETED:
                Toast.makeText(context, "SampleWakeUpBroadcastReceiver: ACTION_BOOT_COMPLETED", Toast.LENGTH_SHORT).show();
                MainApplication mainApplication = (MainApplication) context.getApplicationContext();
                AltBeaconUtil.startBeaconMonitor(mainApplication.getBeaconManager(), mainApplication);
                break;
        }
    }
}

File: Sample Apps /Kotlin/SampleAppWakeUp/app/src/main/java/com/example/sampleappwakeup/BootReceiver.kt
package com.example.sampleappwakeup

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import com.example.sampleappwakeup.application.MainApplication
import com.example.sampleappwakeup.util.AltBeaconUtil

class BootReceiver : BroadcastReceiver() {

    private val altBeaconUtil = AltBeaconUtil()
    override fun onReceive(context: Context?, intent: Intent?) {
        when(intent?.action){
            Intent.ACTION_BOOT_COMPLETED ->{
                Toast.makeText(context,"SampleWakeUpBroadcastReceiver: ACTION_BOOT_COMPLETED",Toast.LENGTH_SHORT).show()
                val mainApplication: MainApplication = context!!.applicationContext as MainApplication
                altBeaconUtil.startBeaconMonitor(mainApplication.getBeaconManager(),mainApplication)
            }
        }
    }
}